tests/falcon/test_vulnerable.py

```diff
@@ -16,54 +19,28 @@ def client():
     return testing.TestClient(app)
 
 
-def test_home(client):
-    response = client.simulate_get("/vulnpy")
-    assert response.status_code == 200
-
-
-def test_cmdi_basic(client):
-    response = client.simulate_get("/vulnpy/cmdi")
-    assert response.status_code == 200
-
-
-def test_cmdi_os_system_normal(client):
-    response = client.simulate_get(
-        "/vulnpy/cmdi/os-system", params={"user_input": "echo attack"}
-    )
+@parametrize_root
+def test_root_views(client, view_name):
+    if view_name == "home":
+        response = client.simulate_get("/vulnpy")
+    else:
+        response = client.simulate_get("/vulnpy/{}".format(view_name))
     assert response.status_code == 200
 
 
-def test_cmdi_os_system_bad_command(client):
-    response = client.simulate_get(
-        "/vulnpy/cmdi/os-system", params={"user_input": "foo"}
-    )
-    assert response.status_code == 200
+@parametrize_triggers
+@pytest.mark.parametrize("request_method", ["simulate_get"])
+def test_trigger(client, request_method, view_name, trigger_name):
+    get_or_post = getattr(client, request_method)
 
+    data = DATA.get(view_name)
 
-def test_cmdi_os_system_invalid_input(client):
-    response = client.simulate_get(
-        "/vulnpy/cmdi/os-system", params={"ignored_param": "bad"}
-    )
-    assert response.status_code == 200
-
+    if view_name == "unsafe_code_exec":
+        data = quote(data)
```

Comment:
Not totally sure but there's probably no harm in calling `quote` the data no matter what.

Reply:
and just a note that  the flask one has to stay like this bc it doesn't work to use quote 
```
    if view_name == "unsafe_code_exec":
        data = "'{}'".format(data)
```